frontend/src/components/Field/YesNo/YesNo.tsx

```diff
@@ -0,0 +1,169 @@
+  children: React.ReactNode
+
+  /**
+   * Variant of the option for styling to be used for styling.
+   */
+  variant: 'left' | 'right'
+
+  /**
+   * Color scheme of the component to render. Defaults to `primary`.
+   */
+  colorScheme?: FieldColorScheme
+}
+
+const YesNoOption = forwardRef<YesNoOptionProps, 'input'>(
+  ({ children, variant, colorScheme = 'primary', ...props }, ref) => {
+    const style = useStyleConfig(YESNO_THEME_KEY, { variant, colorScheme })
+
+    const { getInputProps, getCheckboxProps } = useRadio({
+      ...props,
+      isDisabled: props.disabled,
+    })
+    const input = getInputProps({}, ref)
+
+    const checkbox = getCheckboxProps({})
+
+    const handleSelect = useCallback(() => {
+      if (props.isChecked && input.onChange) {
+        // eslint-disable-next-line @typescript-eslint/no-extra-semi
+        ;(input.onChange as UseRadioGroupReturn['onChange'])('')
+      }
+    }, [input.onChange, props.isChecked])
+
+    const handleSpacebar = useCallback(
+      (e: KeyboardEvent<HTMLInputElement>) => {
+        if (e.key !== ' ') return
+        if (props.isChecked) {
+          e.preventDefault()
+          handleSelect()
+        }
+      },
+      [handleSelect, props.isChecked],
+    )
+
+    return (
+      <Box
+        as="label"
+        w="100%"
+        zIndex={props.isChecked ? 1 : 'initial'}
+        _active={{ zIndex: 1 }}
+        _focusWithin={{ zIndex: 1 }}
+      >
+        <input {...input} onClick={handleSelect} onKeyDown={handleSpacebar} />
+        <VisuallyHidden>
+          "{children}" option {props.isChecked ? 'selected' : 'unselected'}
+        </VisuallyHidden>
+        <Box {...checkbox} __css={style}>
+          {children}
+        </Box>
+      </Box>
+    )
+  },
+)
+
+export const YesNo = forwardRef<YesNoProps, 'input'>(
+  ({ isDisabled, colorScheme = 'primary', ...props }, ref) => {
+    const { getRootProps, getRadioProps } = useRadioGroup(props)
+
+    const groupProps = getRootProps()
+    const [noProps, yesProps] = useMemo(() => {
+      const baseProps = {
+        enterKeyHint: '',
+        disabled: isDisabled,
+        id: props.name,
+      }
+      return [
+        getRadioProps({
+          value: 'no',
+          ...baseProps,
+        }),
+        getRadioProps({
+          value: 'yes',
+          ...baseProps,
+        }),
+      ]
+    }, [getRadioProps, isDisabled, props.name])
+
+    return (
+      <HStack spacing="-px" {...groupProps}>
+        {/* Ref is set here so any errors can focus this input */}
+        <YesNoOption
+          variant="left"
+          colorScheme={colorScheme}
+          {...noProps}
+          ref={ref}
+        >
+          <Icon
+            display={['none', 'none', 'initial']}
+            as={BiX}
+            fontSize="1.5rem"
+            mr="0.5rem"
+            aria-hidden
+          />
+          No
+        </YesNoOption>
+        <YesNoOption variant="right" colorScheme={colorScheme} {...yesProps}>
```

Comment:
Just wondering, why is the `ref` only passed to the `no` option but not the `yes` option?

Reply:
Since this is a radio component, we only want the first ref to be acted on by react hook form. If we pass the same ref to both components, on any form error, the last ref (the "yes" option) instead of the "no" option. You can try it in the Playground story, set isRequired to true and submit without selecting any option

Reply:
ah okay thanks!

Reply:
@karrui can we document this pls? I don't understand even after you've explained. also based on your explanation, is this implementation coupled to react-hook-form?

Reply:
Thanks for all the questions, after changes and experimentation, the same refs can be passed to both options and any errors will still target the first option (i.e. the "No" option).

re @mantariksh: after the change, the implementation is the standard implementation for radio groups (afaik)